DatadogCore/Sources/Core/Storage/Directories.swift

```diff
@@ -31,6 +31,7 @@ internal struct CoreDirectory {
     /// - Returns: The Feature's directories
     func getFeatureDirectories(forFeatureNamed name: String) throws -> FeatureDirectories {
         return FeatureDirectories(
+            root: try coreDirectory.createSubdirectory(path: "\(name)"),
```

Comment:
question/ The `root` doesn't seem to be used in production code. Is it necessary?

I only see `root` being used in tests (to access DS directory), which introduces large inconsistency:
- the real path to feature DS is [created here](https://github.com/DataDog/dd-sdk-ios/blob/72c95e5edadc6c3bfc2dbaf563f2e92854c7e5cd/DatadogCore/Sources/Core/DataStore/FeatureDataStore.swift#L38);
- and here, the `root` targets exactly the same path without keeping both in sync.

We may benefit from having `dataStorePath` defined in `FeatureDirectories` and used by `FeatureDataStore`. This will centralize the DS path management and solve the inconsistency. WDYT @ganeshnj ?

Reply:
On a high level it makes sense but current `getFeatureDirectories` API is throws which makes it harder to integrate with existing system.

However, I made some changes and introduced a new API which is similar that returns the path of the data store which can be used to create dir later on. This also aligns perfectly with the approach we took where data store dir is created on demand.